frontend/packages/kubevirt-plugin/src/components/vms/vm-events.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+
+import { getName, getNamespace } from '@console/shared/src';
+import { PodModel } from '@console/internal/models';
+import { ResourcesEventStream } from '@console/internal/components/events';
+import { VMTabProps } from './types';
+import {
+  VirtualMachineInstanceMigrationModel,
+  VirtualMachineInstanceModel,
+  VirtualMachineModel,
+} from '../../models';
+import { VIRT_LAUNCHER_POD_PREFIX } from '../../constants/vm';
+
+export const VmEvents: React.FC<VMTabProps> = ({ obj: vm }) => {
+  const vmName = getName(vm);
+  const vmNamespace = getNamespace(vm);
+  const vmCompareObject = {
+    name: vmName,
+    namespace: vmNamespace,
+  };
+
+  const vmFilter = (obj) => _.isMatch(obj, { ...vmCompareObject, kind: VirtualMachineModel.kind });
+  const vmiFilter = (obj) =>
+    _.isMatch(obj, { ...vmCompareObject, kind: VirtualMachineInstanceModel.kind });
+
+  const launcherPodFilter = ({ kind, namespace, name }) =>
+    kind === PodModel.kind &&
```

Comment:
I'd extract these functions as utilities, like `isVM`, `isVMI`, `isLauncherPod`, `isImporterPod` and `isMigration`.

Reply:
extract into its own filter functions